src/PcieTypes.bsv

```diff
@@ -0,0 +1,227 @@
+import AxiStreamTypes :: *;
+
+typedef 8 BYTE_BITS
+typedef TMul#(4, BYTE_BITS) DWORD_BITS;
+
+typedef 512 PCIE_TDATA_WIDTH;
+typedef 64  PCIE_TDATA_BYTES;
+typedef 16  PCIE_TDATA_DWORDS;
+// Indicate DWORD valid of tDATA
+typedef PCIE_TDATA_DWORDS PCIE_TKEEP_WIDTH;  
+// tUser width vary among RR, RC, CR and CC
+typedef 183 PCIE_COMPLETER_REQUEST_TUSER_WIDTH;
+typedef 81  PCIE_COMPLETER_COMPLETE_TUSER_WIDTH;
+typedef 137 PCIE_REQUESTER_REQUEST_TUSER_WIDTH;
+typedef 161 PCIE_REQUESTER_COMPLETE_TUSER_WIDTH;
+
+// PcieTlpCtl**: SideBand Signals delivered in tUser defined by PG213
+typedef 8  PCIE_TLP_FIRST_BE_WIDTH;
+typedef 8  PCIE_TLP_LAST_BE_WIDTH;
+typedef Bit#(PCIE_TLP_FIRST_BE_WIDTH)       PcieTlpCtlFirstByteEn;
+typedef Bit#(PCIE_TLP_LAST_BE_WIDTH)        PcieTlpCtlLastByteEn;
+typedef PCIE_TDATA_BYTES PCIE_TLP_BYTE_EN_WIDTH;
+typedef Bit#(PCIE_TLP_BYTE_EN_WIDTH)        PcieTlpCtlByteEn;
+typedef 2  PCIE_TLP_ISSOP_WIDTH;
+typedef 2  PCIE_TLP_ISSOP_PTR_WIDTH;
+typedef Bit#(PCIE_TLP_ISSOP_WIDTH)          PcieTlpCtlIsSop;
+typedef Bit#(PCIE_TLP_ISSOP_PTR_WIDTH)      PcieTlpCtlIsSopPtr;
+typedef 2  PCIE_TLP_ISEOP_WIDTH;
+typedef 4  PCIE_TLP_ISEOP_PTR_WIDTH;
+typedef Bit#(PCIE_TLP_ISEOP_WIDTH)          PcieTlpCtlIsEop;
+typedef Bit#(PCIE_TLP_ISEOP_PTR_WIDTH)      PcieTlpCtlIsEopPtr;
+typedef 2  PCIE_TPH_PRESENT_WIDTH;
+typedef 4  PCIE_TPH_TYPE_WIDTH;
+typedef 16 PCIE_TPH_STTAG;
+typedef 2  PCIE_TPH_INDIRECT_TAGEN_WIDTH;
+typedef Bit#(PCIE_TPH_PRESENT_WIDTH)        PcieTlpCtlTphPresent;
+typedef Bit#(PCIE_TPH_TYPE_WIDTH)           PcieTlpCtlTphType;
+typedef Bit#(PCIE_TPH_STTAG)                PcieTlpCtlTphSteeringTag;
+typedef Bit#(PCIE_TPH_INDIRECT_TAGEN_WIDTH) PcieTlpCtlTphIndirectTagEn;
+typedef 64 PCIE_TLP_PARITY              
+typedef Bit#(PCIE_TLP_PARITY)               PcieTlpCtlParity;
+typedef 4  PCIE_TLP_ADDR_OFFSET_WIDTH;
+typedef Bit#(PCIE_TLP_ADDR_OFFSET_WIDTH)    PcieTlpCtlAddrOffset;
+typedef 6  PCIE_TLP_SEQ_NUM_WIDTH;
+typedef Bit#(PCIE_TLP_SEQ_NUM_WIDTH)        PcieTlpCtlSeqNum;
+typedef 4  PCIE_TLP_RC_ISSOP_WIDTH;
+typedef Bit#(PCIE_TLP_RC_ISSOP_WIDTH)       PcieTlpCtlIsSopRC;
+typedef 4  PCIE_TLP_RC_ISEOP_WIDTH;
+typedef Bit#(PCIE_TLP_RC_ISEOP_WIDTH)       PcieTlpCtlIsEopRC;
+// Signals the start of a new TLP, 6 bit.
+typedef struct {
+    PcieTlpCtlIsSop                 isSop;
```

Comment:
for the name isXXXX, we would think that it is a Bool value, but here, two channels flag are combined into a single struct field.
should we split it into two Bool fields? what's your opinion?

Reply:
This struct will be flattened into a wide _wire_ and passed to the verilog interface. Splitting it may increases the bsv code readability, but causes confusion with the bit meaning in the PG213 document.
![image](https://github.com/BIGWJZ/blue-dmac/assets/49031111/822d1ed9-04cc-449b-8cdf-20be82a55a4e)